sequence_processing_pipeline/NuQCJob.py

```diff
@@ -116,6 +118,63 @@ def __init__(self, fastq_root_dir, output_path, sample_sheet_path,
 
         self._validate_project_data()
 
+    def contains_bx_metadata(self):
```

Comment:
My concern with this is that we are only checking for BX and this could change in the future, no? Also, I'm wondering why we need this vs. having a flag (or something like a flag) of when the pipeline is processing a TellSeq or any other type that has sequence metadata.

Reply:
Why do we need to do this check?

Reply:
Since we know whether we are processing tellseq data, we know the files have these tags. Right?

Reply:
I'm okay with that assumption if you guys are. When I was developing the change to handle the BX metadata, it seemed like minimap2 was okay with passing in an empty list of tags or a list of tags that weren't present in the sequence_id line. Hence I was always passing 'BX' to minimap2. This turned out to not be true when we were running non TellSeq jobs. This is the production hack to make all situations work.

I'm pretty sure that BX will always be present in the integrated fastq file output of a TellSeq job. Similarly, it's not likely to be in a non-TellSeq job since the tags don't appear to be standardized. From what I've seen so far, we'd need to confirm that we can parse a new tag correctly if we want to handle a new one.

I'll go ahead and make the changes.

Reply:
It will now be qp_knight_lab_processing's job to instantiate NuQCJob with the 'BX' parameter for tellseq jobs.